spec/features/staff/users_can_view_an_activity_as_xml_spec.rb

```diff
@@ -37,15 +36,26 @@
 
       it "contains the extending organisation XML" do
         visit organisation_activity_path(organisation, activity, format: :xml)
-        expect(xml.at("iati-activity/participating-org[@role = '3']/@ref").text).to eq(activity.extending_organisation_reference)
-        expect(xml.at("iati-activity/participating-org[@role = '3']/@type").text).to eq(activity.extending_organisation_type)
-        expect(xml.at("iati-activity/participating-org[@role = '3']/narrative").text).to eq(activity.extending_organisation_name)
+        expect(xml.at("iati-activity/participating-org[@role = '3']/@ref").text).to eq(activity.extending_organisation.iati_reference)
+        expect(xml.at("iati-activity/participating-org[@role = '3']/@type").text).to eq(activity.extending_organisation.organisation_type)
+        expect(xml.at("iati-activity/participating-org[@role = '3']/narrative").text).to eq(activity.extending_organisation.name)
       end
 
       it "contains the transaction XML" do
+        transaction = create(:transaction, activity: activity)
         visit organisation_activity_path(organisation, activity, format: :xml)
         expect(xml.at("iati-activity/transaction/@ref").text).to eq(transaction.reference)
       end
+
+      it "contains the budget XML" do
```

Comment:
Given the duplication of the tests we perform I was thinking of a generic activity spec that test how the pieces fit together and then in the contexts for each level we could test what makes that level different from the generic, what do you think?

Reply:
Good call, there is tons of duplication here!